packages/pwa/app/utils/url.js

```diff
@@ -102,16 +103,23 @@ export const productUrlBuilder = (product) => encodeURI(`/product/${product.id}`
 export const searchUrlBuilder = (searchTerm) => `/search?q=${searchTerm}`
 
 /**
- * Returns a URL using the new locale.
+ * Returns a URL pathname using the new locale.
  *
- * @param previousLocale
- * @param newLocale
+ * @param {string} newLocale
+ * @param {Object} [options]
+ * @param {Object} options.location - original location (similar value to `window.location`)
  * @returns {string}
  */
-export const buildUrlLocale = (previousLocale, newLocale) => {
+export const buildUrlLocale = (newLocale, options = {}) => {
+    const location = options.location
+        ? options.location
+        : typeof window !== 'undefined'
+        ? window.location
+        : useLocation()
     const params = new URLSearchParams(location.search)
     params.delete('refine')
-    return location.pathname === HOME_HREF
+
+    return isHomepage(location.pathname)
```

Comment:
There was a bug where `/` is considered a homepage but not `/ja-JP`.

Reply:
Where is the bug from?

Reply:
@alexvuong the bug came from a previous PR. The code was checking for only `/` as the homepage URL. But there are other variants that can be considered homepage too, like `/en-US`, `/ja-JP` and so on.